xtask/src/vulnerabilities.rs

```diff
@@ -0,0 +1,293 @@
+            Sanitizer::Memory => "-Zsanitizer=memory -Zsanitizer-memory-track-origins".to_string(),
+            Sanitizer::MemTag => "--Zsanitizer=memtag -Ctarget-feature=\"+mte\"".to_string(),
+            Sanitizer::SafeStack => "-Zsanitizer=safestack".to_string(),
+            Sanitizer::ShadowCallStack => "-Zsanitizer=shadow-call-stack".to_string(),
+            Sanitizer::Thread => "-Zsanitizer=thread".to_string(),
+        }
+    }
+
+    fn cargo_features(&self) -> Vec<&str> {
+        match self {
+            Sanitizer::CFI => vec!["-Zbuild-std", "--target x86_64-unknown-linux-gnu"],
+            _ => vec![],
+        }
+    }
+
+    fn supported_targets(&self) -> Vec<Target> {
+        match self {
+            Sanitizer::Address => vec![
+                Target::Aarch64AppleDarwin,
+                Target::Aarch64UnknownFuchsia,
+                Target::Aarch64UnknownLinuxGnu,
+                Target::X8664AppleDarwin,
+                Target::X8664UnknownFuchsia,
+                Target::X8664UnknownFreebsd,
+                Target::X8664UnknownLinuxGnu,
+            ],
+            Sanitizer::CFI => vec![Target::X8664UnknownLinuxGnu],
+            Sanitizer::HWAddress => {
+                vec![Target::Aarch64LinuxAndroid, Target::Aarch64UnknownLinuxGnu]
+            }
+            Sanitizer::KCFI => vec![
+                Target::Aarch64LinuxAndroid,
+                Target::Aarch64UnknownLinuxGnu,
+                Target::X8664LinuxAndroid,
+                Target::X8664UnknownLinuxGnu,
+            ],
+            Sanitizer::Leak => vec![
+                Target::Aarch64AppleDarwin,
+                Target::Aarch64UnknownLinuxGnu,
+                Target::X8664AppleDarwin,
+                Target::X8664UnknownLinuxGnu,
+            ],
+            Sanitizer::Memory => vec![
+                Target::Aarch64UnknownLinuxGnu,
+                Target::X8664UnknownFreebsd,
+                Target::X8664UnknownLinuxGnu,
+            ],
+            Sanitizer::MemTag => vec![Target::Aarch64LinuxAndroid, Target::Aarch64UnknownLinuxGnu],
+            Sanitizer::SafeStack => vec![Target::X8664UnknownLinuxGnu],
+            Sanitizer::ShadowCallStack => vec![Target::Aarch64LinuxAndroid],
+            Sanitizer::Thread => vec![
+                Target::Aarch64AppleDarwin,
+                Target::Aarch64UnknownLinuxGnu,
+                Target::X8664AppleDarwin,
+                Target::X8664UnknownFreebsd,
+                Target::X8664UnknownLinuxGnu,
+            ],
+        }
+    }
+
+    // Returns true if the sanitizer is supported by the currently installed targets
+    fn is_target_supported(&self) -> bool {
+        let installed_targets = rustup_get_installed_targets();
+        let supported = self.supported_targets();
+        installed_targets.lines().any(|installed| {
+            supported
+                .iter()
+                .any(|target| target.to_string() == installed.trim())
```

Comment:
Is it necessary to convert `target` into a `String` or we can compare `&str`?

Reply:
I believe so, I use the Display trait implemented for Target variants. Any improvement over this is welcome, do you think of another way to do the comparison ?

Reply:
I do not know if I understood the problem correctly: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1425862775d764bf7a531a8a756b01b4

Reply:
Thank you for the example. Well I can do `.any(|target| target.to_string().eq(installed.trim()))` but that's still allocating a `String`. Since I use the idiomatic Display trait and since this code is not performance critical I believe we can let it like that. What do you think ?

Reply:
If you want to be cleaner, we should parse the string into the enum instead. You would have to add `PartialEq` on the `Target` type, but it would be more robust.